python/ml4ir/applications/ranking/config/parse_args.py

```diff
@@ -25,6 +25,13 @@ def define_args(self):
             "applications/ranking/config/keys.py",
         )
 
+        self.add_argument(
+            "--use_linear_model",
```

Comment:
I was wondering if we should use an enum such as "model_type". This can live like this for now, but we will have to have other bools and deal with conflicts when we come-up with mode then generic vs linear model.

One other way would be to have a param/field "type: RankingModel", "LinearRankingModel" in the feature config yaml.

Any thoughts @lastmansleeping @jakemannix?

Reply:
Yeah, I like the former for now. The problem (maybe, I'm not sure yet) with the latter approach is that a linear model might have multiple different meanings. As in, we can choose to add dropout, other post score layers, maybe even feature function layers, etc. I did try to do the latter but then dropped it for these reasons. I can do the former if you think that's a good idea.

Reply:
This is not correct use of arguments in Python and argparse. As a matter of fact, I believe this will not work, passing something as `bool` does not do True/False as one would expect. Please see here: https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/config/parse_args.py#L256

Also, the most pythonic way to do this to pass with an action store true/false.

@Ullimague argparser support choices out of the box, see here: https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/config/parse_args.py#L62

Reply:
`loss_type` and `scoring_type` can be improved a lot too using these two pointers (all necessary checks will be done by argparser)

Reply:
@balikasg Thanks, I will update to this.